diff --git a/Pipeto/Pipeto/src/commands/unlock_secret_mode.c b/Pipeto/Pipeto/src/commands/unlock_secret_mode.c
index 020d8c2..d47f0b3 100644
--- a/Pipeto/Pipeto/src/commands/unlock_secret_mode.c
+++ b/Pipeto/Pipeto/src/commands/unlock_secret_mode.c
@@ -10,12 +10,25 @@
 #include <string.h>
 #include "pipeto.h"
 
-void unlock_secret_mode()
+static void encrypted_flag(void)
+{
+    unsigned char enc_flag[] = {0xb8, 0x3a, 0x9b, 0x93, 0xd5, 0x0c, 0x19,
+        0x7d, 0x2e, 0x13, 0x49};
+    unsigned char keys[] = {0xc3, 0x7b, 0xdf, 0xde, 0x9c, 0x42, 0x2d, 0x4f,
+        0x1a, 0x21, 0x34};
+    char decrypted[12];
+
+    for (unsigned int i = 0; i < sizeof(enc_flag); i++)
+        decrypted[i] = enc_flag[i] ^ keys[i];
+    decrypted[11] = '\0';
+    printf("%s\n", decrypted);
+}
+
+void unlock_secret_mode(void)
 {
     if (IS_ADMIN) {
         printf("Secret mode unlocked! Welcome, admin.\n");
-        printf("{ADMIN4242}\n");
-    } else {
+        encrypted_flag();
+    } else
         printf("Access denied. You do not have the required privileges.\n");
-    }
 }
